Jenkinsfile

```diff
@@ -37,65 +51,117 @@ void unittests(String version) {
-    Map tasks = [ : ]
-
-    pythonVersion.each {
-        version ->
-            tasks[version] = { unittests(version) }
-    }
-
-    tasks['snyk'] = {
-        stage('snyk') {
-            Snyk snyk = new Snyk(this, bparams)
-            snyk.execute()
-        }
-    }
-
-    tasks['sonarqube'] = {
-        stage('sonarqube') {
-            SonarQube.execute(this, bparams)
-        }
-    }
-
-    tasks['nexusiq'] = {
-        stage('nexusiq') {
-            Nexusiq.execute(this, bparams)
-        }
-    }
-
-    parallel(tasks)
-
-    common.setResultIfNotSet(Constants.JSUCCESS)
-} 
-catch (ex) {
-    common.logException(ex)
-    common.setResultAbortedOrFailure()
-    throw ex
-} 
-finally {
-    common.setResultIfNotSet(Constants.JFAILURE)
-    buildsCommon.notifyPostBuild(bparams)
+	Map tasks = [: ]
+
+	pythonVersion.each {
+		version ->tasks[version] = {
+			unittests(version)
+		}
+	}
+
+	tasks['snyk'] = {
+		stage('snyk') {
+			Snyk snyk = new Snyk(this, bparams)
+			snyk.execute()
+		}
+	}
+
+	tasks['sonarqube'] = {
+		stage('sonarqube') {
+			SonarQube.execute(this, bparams)
+		}
+	}
+
+	tasks['nexusiq'] = {
+		stage('nexusiq') {
+			Nexusiq.execute(this, bparams)
+		}
+	}
+
+	tasks['runPylint'] = {
+		stage('runPylint') {
+			node(Constants.DOCKERNODE) {
+				withContainer(image: "python:3.6-buster", registry: '', inside: '-u root') {
+					try {
+						sh """
+						mkdir reports
+						pip install pylint
+                           			pylint --exit-zero --output-format=parseable --reports=n tenable > reports/pylint_tenable.log
+                           			pylint --exit-zero --output-format=parseable --reports=n tests > reports/pylint_tests.log
+                           			cat reports/pylint_tenable.log
+                           			cat reports/pylint_tests.log
+                        			"""
+					} catch(ex) {
+						throw ex
+					} finally {
+						if (fileExists('reports/pylint_tenable.log')) {
+							//result should available. TODO: to test
+							result = recordIssues(
+							enabledForFailure: true, tool: pyLint(pattern: 'reports/pylint_tenable.log'), unstableTotalAll: 20, failedTotalAll: 30, )
+						}
+						if (fileExists('reports/pylint_test.log')) {
+							//result should available. TODO: to test
+							result = recordIssues(
+							enabledForFailure: true, tool: pyLint(pattern: 'reports/pylint_test.log'), unstableTotalAll: 20, failedTotalAll: 30, )
+						}
+					}
+				}
+			}
+		}
+	}
+
+	parallel(tasks)
+
+	common.setResultIfNotSet(Constants.JSUCCESS)
+	
+	tasks['runPyPI'] = {
+	  stage('runPyPI') {
+	    node(Constants.DOCKERNODE) {
+	      withContainer(image: "python:3.6-buster", registry: '', inside: '-u root') {
+		steps {
+		  try {
+		  sh """
+		        sudo apt-get install jq
+		        pypi_version=`curl -Ls https://pypi.org/pypi/pyTenable/json | jq -r .info.version`
+			current_version=$(python -c "import tenable;print (tenable.__version__)")
+			if [ $pypi_version != $current_version ]
```

Comment:
These are shell variables, so they ned to be escaped with \ or you need to replace the double quotes with single quotes.

Reply:
Do we have the withGithubNotify(context: 'Release') , then at the time of Release , automatically the release function will be executed? I had added that, it says, plugin not found. Can you clarify

Reply:
Do we have withGithubNotify plugin in our environment?